pkg/kubelet/util/format/pod.go

```diff
@@ -63,3 +65,29 @@ func aggregatePods(pods []*api.Pod, handler podHandler) string {
 	}
 	return fmt.Sprintf(strings.Join(podStrings, ", "))
 }
+
+func ExpandConfigMap(configMap *api.ConfigMap, pod *api.Pod) {
+	r := regexp.MustCompile("\\$\\([a-zA-Z0-9.]+\\)")
+	mapping_funcs := map[string]func(*api.Pod) string{
+		"$(status.podIP)":   func(pod *api.Pod) string { return pod.Status.PodIP },
+		"$(status.message)": func(pod *api.Pod) string { return pod.Status.Message },
+		"$(status.phase)":   func(pod *api.Pod) string { return string(pod.Status.Phase) },
+		"$(pod.name)":       func(pod *api.Pod) string { return pod.Name },
+		"$(pod.labels)": func(pod *api.Pod) string {
+			labels := make([]string, 0, len(pod.Labels))
+			for label, value := range pod.Labels {
+				labels = append(labels, label+"="+value)
+			}
+
+			return strings.Join(labels, ",")
+		},
+	}
+	for k := range configMap.Data {
+		vars := r.FindAllString(configMap.Data[k], -1)
+		for _, v := range vars {
+			value := mapping_funcs[v](pod)
+			configMap.Data[k] = strings.Replace(configMap.Data[k], v, value, -1)
```

Comment:
Just for learning, If the pod is restarted for some reason, do we still have the "$(status.podIP)" string in the configmap? (Since the PodIP may change?)